### DISCUSSION TITLE
How to shift text down a little with convert/caption/undercolor?

### DISCUSSION THREAD

**Discussion by user88578** [ImageMagick/ImageMagick#2433] [Help]
I am using this currently

```
convert -background none original.png -undercolor yellow -fill red -stroke black \
          -font NimbusRoman-Italic -gravity center -size 514x200 caption:"Text Here" -composite draw_text.png
```

But the text is butted up right to the top of the yellow banner. How do I make it centered in the undercolor field?

**Comment by fmw42** [comment]
I can confirm. But I think that is just the way the font glyph was defined. I am unaware of a way to shift that down. But perhaps a developer might comment.

**Reply by user88578** [reply]
> I can confirm. But I think that is just the way the font glyph was defined. I am unaware of a way to shift that down. But perhaps a developer might comment.

I saw someone saying they achieved it somehow using `annotate` here:

https://old.reddit.com/r/imagemagick/comments/gcaq7c/adding_text_to_an_image/

I tried it yesterday but failed, didn't understand what code to use for my case tbh.

**Reply by fmw42** [reply]
In that link, they are only appending the text with undercolor above the main image. That is not shifting the text within the undercolor.

**Reply by user88578** [reply]
> In that link, they are only appending the text with undercolor above the main image. That is not shifting the text within the undercolor.

ok, red herring :). Wait for more opinions.

**Comment by snibgo** [comment]
The undercolor fills the "annotation bounding box", and the gap between this and the gylphs probably depends on the font used. If I use "-font Arial", there is a good gap.
![d](https://user-images.githubusercontent.com/33812368/90636795-0c850b80-e223-11ea-9da4-89c36bb2f006.png)
